### PR TITLE
consolidate connector tests

### PR DESCRIPTION
**Pull request by colombod**

### CODE REVIEW

File: src/Microsoft.DotNet.Interactive.Tests/KernelConnectionTestsBase.cs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        _disposables.Dispose();
+    }
+
+    [WindowsFact]
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Interoperability", "CA1416:Validate platform compatibility", Justification = "Test only enabled on windows platforms")]
+    public async Task it_can_reuse_connection_for_multiple_proxy_kernel()
+    {
+        var configuration = CreateConnectionConfiguration();
+       
+        var connector = await CreateConnectorAsync(configuration);
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.Tests/KernelConnectionTestsBase.cs:46]: Should this not be diposable? I noticed that `NamedPipeKernelConnector` creates a `MultiplexingKernelCommandAndEventReceiver` which is disposable, but never disposes it.

File: src/Microsoft.DotNet.Interactive.Tests/StdioConnectionTests.cs [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+    protected override Task<IKernelConnector> CreateConnectorAsync(StdioConnectionTestConfiguration configuration)
+    {
+        var command = new List<string>{configuration.Command};
+        if (configuration.Args?.Length > 0)
+        {
+            command.AddRange(configuration.Args);
+        }
+
+        var connector = new StdIoKernelConnector(command.ToArray(), configuration.WorkingDirectory);
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.Tests/StdioConnectionTests.cs:38]: This constructor should be fixed to differentiate the executable path from the subsequent command line args.

File: src/Microsoft.DotNet.Interactive.Tests/StdioConnectionTests.cs [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+                $"\"{toolPath.FullName}\"",
+                "stdio",
+                "--default-kernel",
+                "csharp",
+            },
+            WorkingDirectory = toolPath.Directory
+        };
+    }
+
+    private FileInfo CopyToTemp(string assemblyLocation)
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.Tests/StdioConnectionTests.cs:63]: We have a `TemporaryDirectory` class with cleanup semantics. Maybe use that here?

File: src/Microsoft.DotNet.Interactive.Tests/StdioConnectionTests.cs [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+    protected override Task<IDisposable> ConnectHostAsync(CompositeKernel remoteKernel, StdioConnectionTestConfiguration configuration)
+    {
+        return Task.FromResult(Disposable.Empty);
+    }
+}
+
+public class StdioConnectionTestConfiguration
+{
+    public string Command { get; set; }
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.Tests/StdioConnectionTests.cs:154]: This name is ambiguous. Maybe ExePath? Or, realistically in this context, this is `DotNetExePath`? Is it ever different? Is it meaningful to make this settable?

File: src/dotnet-interactive-vscode-common/src/interactiveClient.ts [hunk 5]
```diff
@@ -22,7 +22,7 @@ import {
     KernelEventEnvelope,
     KernelEventEnvelopeObserver,
     KernelEventType,
-    KernelTransport,
+    Connector,
```
Comment (jonsequitur) [inline src/dotnet-interactive-vscode-common/src/interactiveClient.ts:25]: Can this name be a bit more specific?

File: src/Microsoft.DotNet.Interactive.Tests/StdioConnectionTests.cs (hunk 3 above)
Comment (jonsequitur) [reply src/Microsoft.DotNet.Interactive.Tests/StdioConnectionTests.cs:63]: Why is this copy needed?

File: src/Microsoft.DotNet.Interactive.Tests/KernelConnectionTestsBase.cs (hunk 1 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive.Tests/KernelConnectionTestsBase.cs:46]: I think that kernelConnectors should be all disposable.

File: src/Microsoft.DotNet.Interactive.Tests/StdioConnectionTests.cs (hunk 4 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive.Tests/StdioConnectionTests.cs:154]: That is the binary to run, trying to model like the process startinfo thing

File: src/dotnet-interactive-vscode-common/src/interactiveClient.ts (hunk 5 above)
Comment (colombod) [reply src/dotnet-interactive-vscode-common/src/interactiveClient.ts:25]: will align that even more to .net api names

File: src/Microsoft.DotNet.Interactive.Tests/StdioConnectionTests.cs (hunk 3 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive.Tests/StdioConnectionTests.cs:63]: using the utility, I am copying this to ensure no lock during local ncrunch / parallel test runs

File: src/Microsoft.DotNet.Interactive.ExtensionLab/SQLiteKernelConnector.cs [hunk 6]
```diff
@@ -22,5 +22,9 @@ public SQLiteKernelConnector(string connectionString)
         {
             ConnectionString = connectionString;
         }
+
+        public void Dispose()
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.ExtensionLab/SQLiteKernelConnector.cs:26]: Why the empty `Dispose` methods on several of these?

File: src/Microsoft.DotNet.Interactive.SqlServer/MsSqlKernelConnector.cs [hunk 7]
```diff
@@ -5,42 +5,45 @@
...
-            CreateDbContext = createDbContext;
-            ConnectionString = connectionString;
-        }
+    public MsSqlKernelConnector(bool createDbContext, string connectionString)
+    {
+        CreateDbContext = createDbContext;
+        ConnectionString = connectionString;
+    }
+
+    public void Dispose()
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.SqlServer/MsSqlKernelConnector.cs:46]: Another.

File: src/Microsoft.DotNet.Interactive.Tests/ConnectDirectiveTests.cs [hunk 8]
```diff
@@ -215,5 +215,10 @@ public Task<Kernel> ConnectKernelAsync(KernelInfo kernelInfo)
         {
             return CreateKernel(kernelInfo);
         }
+
+        public void Dispose()
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.Tests/ConnectDirectiveTests.cs:219]: Another.

File: src/Microsoft.DotNet.Interactive.ExtensionLab/SQLiteKernelConnector.cs (hunk 6 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive.ExtensionLab/SQLiteKernelConnector.cs:26]: the connector for sql and kql doesn't need disposing, the toolprocess is controlled by the kernel

File: src/Microsoft.DotNet.Interactive.Tests/Utility/TemporaryDirectory.cs [hunk 9]
```diff
@@ -4,31 +4,58 @@
 using System;
 using System.IO;
 
-namespace Microsoft.DotNet.Interactive.Tests.Utility
+namespace Microsoft.DotNet.Interactive.Tests.Utility;
+
+public class TemporaryDirectory : IDisposable
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.Tests/Utility/TemporaryDirectory.cs:9]: Is this still needed? If not is there a reason not to use the `DisposableDirectory` class we already have?

File: src/Microsoft.DotNet.Interactive/Connection/NamedPipeKernelConnector.cs [hunk 10]
```diff
@@ -7,47 +7,51 @@
...
-    public class NamedPipeKernelConnector : IKernelConnector
+    private MultiplexingKernelCommandAndEventReceiver? _receiver;
+    private KernelCommandAndEventPipeStreamSender? _sender;
+
+    public string PipeName { get; }
+    public async Task<Kernel> ConnectKernelAsync(KernelInfo kernelInfo)
     {
-        private MultiplexingKernelCommandAndEventReceiver? _receiver;
-        private KernelCommandAndEventPipeStreamSender? _sender;
+        ProxyKernel? proxyKernel;
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive/Connection/NamedPipeKernelConnector.cs:20]: Please use explicit visibility.

File: src/Microsoft.DotNet.Interactive.Tests/Utility/TemporaryDirectory.cs (hunk 9 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive.Tests/Utility/TemporaryDirectory.cs]: removed in favour of the one already present

File: src/Microsoft.DotNet.Interactive/Connection/NamedPipeKernelConnector.cs [hunk 11]
```diff
@@ -7,47 +7,51 @@
...
-                    PipeName,
-                    PipeDirection.InOut,
-                    PipeOptions.Asynchronous, TokenImpersonationLevel.Impersonation);
-
-                await clientStream.ConnectAsync();
-                clientStream.ReadMode = PipeTransmissionMode.Message;
-
-                _receiver = new MultiplexingKernelCommandAndEventReceiver(new KernelCommandAndEventPipeStreamReceiver(clientStream));
-                _sender = new KernelCommandAndEventPipeStreamSender(clientStream);
+            var clientStream = new NamedPipeClientStream(
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive/Connection/NamedPipeKernelConnector.cs:28]: Are we ever disposing this?

File: src/Microsoft.DotNet.Interactive/Connection/NamedPipeKernelConnector.cs (hunk 10 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive/Connection/NamedPipeKernelConnector.cs:20]: is a variable

File: src/Microsoft.DotNet.Interactive/Connection/NamedPipeKernelConnector.cs (hunk 11 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive/Connection/NamedPipeKernelConnector.cs]: now we are :)

**Review by jonsequitur — APPROVED** [review]

**Pull request merged**